node/pkg/notary/notary.go

```diff
@@ -329,7 +331,7 @@ func (n *Notary) blackhole(msg *common.MessagePublication) error {
 	n.mutex.Unlock()
 
 	// When a message is blackholed, it should be removed from the delayed list and database.
-	err := n.removeDelayed(msg)
+	err := n.removeDelayed(msg.MessageID())
```

Comment:
Probably applies to the notary PR rather than this one, but should we attempt to remove from the delayed list + database before adding to the blackholed list? Just in case the removal fails we don't want the message in both places